leaderelection/config.go

```diff
@@ -112,6 +112,14 @@ type ComponentConfig struct {
 	LeaseDuration time.Duration
 	RenewDeadline time.Duration
 	RetryPeriod   time.Duration
+	// LeaseName is a function to customize the lease name given the index i.
+	// If not present, a name in format {Component}.{queue-name}.{i}-of-{Buckets}
+	// will be use.
```

Comment:
We probably need to comment _why_ would we need a custom naming scheme.

Reply:
done.